pkg/didcomm/protocol/didexchange/states.go

```diff
@@ -391,63 +409,152 @@ func (ctx *context) handleInboundRequest(request *Request, options *options,
 	logger.Debugf("handling request: %+v", request)
 
-	reqConn, err := getRequestConnection(request)
+	requestConnection, err := getRequestConnection(request)
 	if err != nil {
 		return nil, nil, fmt.Errorf("extracting connection data from request: %w", err)
 	}
 
-	requestDidDoc, err := ctx.resolveDidDocFromConnection(reqConn)
+	requestDidDoc, err := ctx.resolveDidDocFromConnection(requestConnection)
 	if err != nil {
 		return nil, nil, fmt.Errorf("resolve did doc from exchange request connection: %w", err)
 	}
 
-	logger.Debugf("requestDIDDoc: %+v", requestDidDoc)
-
 	// get did document that will be used in exchange response
 	// (my did doc)
-	responseDidDoc, connection, err := ctx.getDIDDocAndConnection(getPublicDID(options), getRouterConnections(options))
+	responseDidDoc, responseConnection, err := ctx.getDIDDocAndConnection(
+		getPublicDID(options), getRouterConnections(options))
 	if err != nil {
-		return nil, nil, err
+		return nil, nil, fmt.Errorf("get response did doc and connection: %w", err)
 	}
 
-	// prepare connection signature
-	encodedConnectionSignature, err := ctx.prepareConnectionSignature(connection, request.Thread.PID)
+	senderVerKey, err := recipientKey(responseDidDoc)
+	if err != nil {
+		return nil, nil, fmt.Errorf("handle inbound request: %w", err)
+	}
+
+	if ctx.doACAPyInterop {
+		// Interop: aca-py issue https://github.com/hyperledger/aries-cloudagent-python/issues/1048
+		responseDidDoc, err = convertPeerToSov(responseDidDoc)
+		if err != nil {
+			return nil, nil, fmt.Errorf("converting my did doc to a 'sov' doc for response message: %w", err)
+		}
+	}
+
+	response, err := ctx.prepareResponse(request, responseDidDoc, responseConnection)
+	if err != nil {
+		return nil, nil, fmt.Errorf("preparing response: %w", err)
+	}
+
+	connRec.TheirDID = requestConnection.DID
+	connRec.MyDID = responseConnection.DID
+	connRec.TheirLabel = request.Label
+
+	destination, err := service.CreateDestination(requestDidDoc)
 	if err != nil {
 		return nil, nil, err
 	}
 
+	if len(destination.MediaTypeProfiles) > 0 {
+		connRec.MediaTypeProfiles = destination.MediaTypeProfiles
+	}
+
+	// send exchange response
+	return func() error {
+		return ctx.outboundDispatcher.Send(response, senderVerKey, destination)
+	}, connRec, nil
+}
+
+func (ctx *context) prepareResponse(request *Request, responseDidDoc *did.Doc,
+	responseConnection *Connection) (*Response, error) {
 	// prepare the response
 	response := &Response{
 		Type: ResponseMsgType,
 		ID:   uuid.New().String(),
 		Thread: &decorator.Thread{
 			ID: request.ID,
 		},
-		ConnectionSignature: encodedConnectionSignature,
 	}
 
-	connRec.TheirDID = reqConn.DID
-	connRec.MyDID = connection.DID
-	connRec.TheirLabel = request.Label
+	if request.Thread != nil {
+		response.Thread.PID = request.Thread.PID
+	}
 
-	destination, err := service.CreateDestination(requestDidDoc)
+	// TODO followup: bring this out from under doACAPYInterop, and instead under an AIP 2.0 flag
```

Comment:
we won't have a flag for AIP 2 - we only care about AIP2+.

Reply:
fixed comment